onnxruntime/core/providers/cuda/shared_inc/fpgeneric.h

```diff
@@ -25,12 +25,14 @@ inline cublasStatus_t cublasGemmHelper(cublasHandle_t handle, cublasOperation_t
   return cublasDgemm(handle, transa, transb, m, n, k, alpha, A, lda, B, ldb, beta, C, ldc);
 }
 inline cublasStatus_t cublasGemmHelper(cublasHandle_t handle, cublasOperation_t transa, cublasOperation_t transb, int m, int n, int k, const half* alpha, const half* A, int lda, const half* B, int ldb, const half* beta, half* C, int ldc) {
+  cublasSetMathMode(handle, CUBLAS_TENSOR_OP_MATH);
   // This does true FP16 computation which is slow for non-Volta GPUs
-  //return cublasHgemm(handle, transa, transb, m, n, k, alpha, A, lda, B, ldb, beta, C, ldc);
+  if (onnxruntime::cuda::DeviceProp().GetDeviceProps().major >= 7) {
+    return cublasHgemm(handle, transa, transb, m, n, k, alpha, A, lda, B, ldb, beta, C, ldc);
+  }
   // This does pseudo FP16 computation (input/output in fp16, computation in fp32)
   float h_a = onnxruntime::math::halfToFloat(*reinterpret_cast<const uint16_t*>(alpha));
```

Comment:
How about removing the pseudo FP16 path and use cublasHGemm only? 

It is probably slower for some GPUs, but user should run FP32 instead of FP16 on those GPUs that does not support FP16 well.